src/web/assets/htmx/src/htmx.js

```diff
@@ -9,19 +9,22 @@ htmx.defineExtension('craft-cp', {
                 break;
         }
     },
-
     configureRequest: function(evt) {
         // Add the standard Craft headers
         Object.assign(evt.detail.headers, Craft._actionHeaders());
     },
 
-    onLoad: function(evt) {
-        if (evt.detail.elt === document.body) {
+    // The best place to do this, until an event like `htmx:newContent` is introduced.
+    transformResponse: function(text, xhr, elt) {
+        const parser = new DOMParser();
+        const doc = parser.parseFromString(text, "text/html");
+
+        if (doc.body === document.body) {
             return;
         }
 
-        const allHeadHtml = evt.detail.elt.querySelectorAll("template.hx-head-html");
-        const allBodyHtml = evt.detail.elt.querySelectorAll("template.hx-body-html");
+        const allHeadHtml = doc.querySelectorAll("template.hx-head-html");
+        const allBodyHtml = doc.querySelectorAll("template.hx-body-html");
 
         for (let i = 0; i < allHeadHtml.length; i++) {
             const headHtml = allHeadHtml[i].innerHTML;
```

Comment:
Maybe we should be using https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent

as: 

> textContent gets the content of all elements, including [<script>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) and [<style>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style) elements. In contrast, innerText only shows "human-readable" elements.

Reply:
Sorry just realized `innerHTML` was being used, not innerText. Ignore.